src/Agent/NewRelic/Agent/Core/Api/AgentApiImplementation.cs

```diff
@@ -381,16 +384,21 @@ private void AddUserAttributeToCurrentTransaction(string key, object value)
         /// category defaults to "Custom". </param>
         /// <param name="name">	    The name of the transaction starting with a forward slash.  example:
         /// /store/order Only the first 1000 characters are retained. </param>
-        public void SetTransactionName(string category, string name)
-
+        public void SetTransactionName(string? category, string name)
         {
             name = name ?? throw new ArgumentNullException(nameof(name));
 
             using (new IgnoreWork())
             {
+                category = category != null
+                    ? category.Trim()
+                    : string.Empty;
+
                 // Default to "Custom" category if none provided
-                if (string.IsNullOrEmpty(category?.Trim()))
+                if (string.IsNullOrEmpty(category))
```

Comment:
Just use `IsNullOrWhitespace`. Probably avoids a string allocation. If you still need to check for null that's fine.